nats/src/kv.rs

```diff
@@ -781,6 +781,9 @@ impl Iterator for Keys {
 
     fn next(&mut self) -> Option<Self::Item> {
         loop {
+            if self.subscription.0.num_pending == 0 {
```

Comment:
should we keep a counter of how many we've received and break out here? i imagine some overlap with self.done break tho